cpu/xive.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+#
+# Copyright: 2021 IBM
+# Author: Kalpana Shetty <kalshett@in.ibm.com>
+
+import os
+import re
+
+from avocado import Test
+from avocado.utils import genio
+
+
+class XIVE(Test):
+
+    '''
+    XIVE test cases
+
+    :avocado: tags=cpu,power
+    '''
+
+    def setUp(self):
+        if "ppc" not in os.uname()[4]:
+            self.cancel("Supported only on Power Systems")
+
+        cpu_info = genio.read_file("/proc/cpuinfo")
```

Comment:
why not we can using `sikpUnless`  class decorator 
code can be like 
```
IS_POWER10= 'Power10' in open('/proc/cpuinfo', 'r').read()
IS_POWER9 = 'POWER9' in open('/proc/cpuinfo', 'r').read()
 @skipUnless( IS_POWER10 | IS_Power9 , "Unsupported processor family")
 
 ```

Reply:
This for generic power platform. So I don't think it really required to use "skipUnless", we do have other Power systems as well POWER8 for example. So I will not change this.

Reply:
sure make sense

Reply:
may we can have skipUnless in other way 
IS_POWER8 = 'POWER8'  not in  open('/proc/cpuinfo', 'r').read()
@skipUnless(! IS_POWER8 , "Unsupported processor family")